onnxruntime/python/onnxruntime_validation.py

```diff
@@ -56,3 +56,95 @@ def check_distro_info():
+    version = ''
+    cuda_version = ''
+
+    if has_ortmodule:
+        try:
+            # collect onnxruntime package name, version, and cuda version
+            from .build_and_package_info import package_name
+            from .build_and_package_info import __version__ as version
+
+            try:
+                from .build_and_package_info import cuda_version
+            except: # noqa
+                pass
+
+            if cuda_version:
+                # collect cuda library build info. the library info may not be available
+                # when the build environment has none or multiple libraries installed
+                try:
+                    from .build_and_package_info import cudart_version
+                except: # noqa
+                    warnings.warn('WARNING: failed to get cudart_version from onnxruntime build info.')
+                    cudart_version = None
+
+                def print_build_package_info():
+                    warnings.warn('onnxruntime training package info: package_name: %s' % package_name)
+                    warnings.warn('onnxruntime training package info: __version__: %s' % version)
+                    warnings.warn('onnxruntime training package info: cuda_version: %s' % cuda_version)
+                    warnings.warn('onnxruntime build info: cudart_version: %s' % cudart_version)
+
+                # collection cuda library info from current environment.
+                from onnxruntime.capi.onnxruntime_collect_build_info \
+                    import find_cudart_versions, find_cudnn_supported_cuda_versions
+                local_cudart_versions = find_cudart_versions(build_env=False, build_cuda_version=cuda_version)
+                if cudart_version and cudart_version not in local_cudart_versions:
+                    print_build_package_info()
+                    warnings.warn('WARNING: failed to find cudart version that matches onnxruntime build info')
+                    warnings.warn('WARNING: found cudart versions: %s' % local_cudart_versions)
+
+                # https://docs.nvidia.com/deeplearning/cudnn/support-matrix/index.html
+                # https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnGetCudartVersion
+                # it is tricky to validate local cudnn support.
+                # There is a case when onnxruntime-training is built with Cuda 11.1.
+                # It however works with cudnn 8.0.3 (8003) which by cudnnGetCudartVersion and
+                # cudnn support-matrix, only support Cuda 11.0.
+                # to avoid false warning, we only post a message if there is no cudnn installed at all,
+                # hoping meanful error message is posted when runtime does not find a compatible cudnn library.
+                local_cudnn_supported_cuda_versions = find_cudnn_supported_cuda_versions(build_env=False)
+                if len(local_cudnn_supported_cuda_versions) == 0:
+                    warnings.warn_build_package_info()
```

Comment:
I couldn't find a reference to `warn_build_package_info()`... is this code path tested?

Reply:
good finding - this is still code-in-progress as I do not get clear understanding of how cudnn versions/cuda support works. my experiment/observation does match cudnn docs.
as a result and suggested by @KeDengMS cudnn validation is removed.